clients/spark/core/src/main/scala/io/treeverse/clients/StorageUtils.scala

```diff
@@ -26,10 +40,45 @@ object StorageUtils {
     def uriToStorageAccountName(storageNsURI: URI): String = {
       storageNsURI.getHost.split('.')(0)
     }
+
+    def concatKeysToStorageNamespace(keys: Seq[String], storageNamespace: String): Seq[String] = {
+      val addSuffixSlash =
+        if (storageNamespace.endsWith("/")) storageNamespace else storageNamespace.concat("/")
+
+      if (keys.isEmpty) return Seq.empty
+      keys
+        .map(x => addSuffixSlash.concat(x))
+        .map(x => x.getBytes(Charset.forName("UTF-8")))
+        .map(x => new String(x))
+    }
   }
 
   object S3 {
     val S3MaxBulkSize = 1000
     val S3NumRetries = 1000
+
+    def concatKeysToStorageNamespace(keys: Seq[String], storageNamespace: String): Seq[String] = {
+      val addSuffixSlash =
+        if (storageNamespace.endsWith("/")) storageNamespace else storageNamespace.concat("/")
+      val snPrefix =
+        if (addSuffixSlash.startsWith("/")) addSuffixSlash.substring(1) else addSuffixSlash
+
+      if (keys.isEmpty) return Seq.empty
+      keys.map(x => snPrefix.concat(x))
+    }
+
+    def concatKeysToStorageNamespacePrefix(
```

Comment:
If I understand correctly, the difference between this function and the one above is that this one trims the "s3://<bucket name>" prefix. The names are confusing - not sure why it's called "concatKeysToStorageNamespace**Prefix**".

Not sure it even requires a separate function. I would add this ability as an option to the function.